utils/ssh.py

```diff
@@ -208,13 +225,28 @@ def run_rake_command(self, command, timeout=RUNCMD_TIMEOUT):
 
     def put_file(self, local_file, remote_file='.', **kwargs):
         logger.info("Transferring local file %s to remote %s", local_file, remote_file)
-        return SCPClient(self.get_transport(), progress=self._progress_callback).put(
-            local_file, remote_file, **kwargs)
+        if self.is_container:
+            tempfilename = '/share/temp_{}'.format(fauxfactory.gen_alpha())
+            scp = SCPClient(self.get_transport(), progress=self._progress_callback).put(
+                local_file, tempfilename, **kwargs)
+            self.run_command('mv {} {}'.format(tempfilename, remote_file))
```

Comment:
Exactly. Basically one additional step with a random name.

The docs I pushed clarify the prerequisities for the VM that hosts the docker container.